#! /bin/sh -e
## 02_quit_blend_in_homedir.dpatch by Masayuki Hatta <mhatta@debian.org>
##
## All lines beginning with \`## DP:' are a description of the patch.
## DP: quit.blend shoule be created in the user's homedir, not /tmp.

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
        -patch) patch -f --no-backup-if-mismatch --dry-run -p1 < $0 && patch -f --no-backup-if-mismatch -p1 < $0
;;
        -unpatch) patch -f --no-backup-if-mismatch -R -p1 < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

diff -urN blender-2.37a.orig/source/blender/src/usiblender.c blender-2.37a/source/blender/src/usiblender.c
--- blender-2.37a.orig/source/blender/src/usiblender.c	2005-09-07 14:16:49.000000000 +0900
+++ blender-2.37a/source/blender/src/usiblender.c	2005-09-07 14:17:18.000000000 +0900
@@ -153,9 +153,7 @@
 	}
 	if(U.mixbufsize==0) U.mixbufsize= 2048;
 	if (BLI_streq(U.tempdir, "/")) {
-		char *tmp= getenv("TEMP");
-		
-		strcpy(U.tempdir, tmp?tmp:"/tmp/");
+		strcpy(U.tempdir, BLI_gethome());
 	}
 	if (U.savetime <= 0) {
 		U.savetime = 1;
